is/utils/io.hpp

```diff
@@ -1,10 +1,64 @@
 #pragma once
 
 #include <string>
+#include <fstream>
 #include "google/protobuf/message.h"
 #include "status.hpp"
+#include "boost/optional.hpp"
+
+namespace google {
+namespace protobuf {
+namespace io {
+class ZeroCopyOutputStream;
+class CodedOutputStream;
+class CodedInputStream;
+}
+}
+}
 
 namespace is {
 wire::Status load(std::string const& filename, google::protobuf::Message* message);
 wire::Status save(std::string const& filename, google::protobuf::Message const& message);
+
+struct ProtobufWriter {
+  std::ofstream file;
+  google::protobuf::io::ZeroCopyOutputStream *raw_output;
+  google::protobuf::io::CodedOutputStream *coded_output;
+
+  ProtobufWriter(std::string const &filename);
+  ~ProtobufWriter();
+  void close();
+
+  template <typename T> bool insert(T const &message) {
```

Comment:
Maybe change the signature to return Status? And return a FAILED_PRECONDITION if we failed to open the file before, otherwise throw at the class constructor...